stubs/docutils/docutils/nodes.pyi

```diff
@@ -1,15 +1,73 @@
+import xml.dom.minidom
 from _typeshed import Self
 from collections.abc import Iterable
-from typing import Any, overload
+from types import ModuleType
+from typing import Any, Callable, Generator, TypeVar, overload
+from typing_extensions import Literal
 
 from docutils.transforms import Transformer
 
+class NodeVisitor:
+    def __init__(self, document: document): ...
+    def __getattr__(self, __name: str) -> Any: ...  # incomplete
+
+_N = TypeVar("_N", bound=Node)
+
 class Node:
     parent: Node | None
     source: str | None
     line: int | None
     document: document | None
-    def __getattr__(self, __name: str) -> Any: ...  # incomplete
+    def __bool__(self) -> Literal[True]: ...
+    def asdom(self, dom: ModuleType | None = ...) -> xml.dom.minidom.Element: ...
+    # ModuleType is inaccurate, you cannot pass any module, but it's the best we can express
```

Comment:
```suggestion
    # ModuleType is inaccurate, you cannot pass any module, but it's the best we can express
    def asdom(self, dom: ModuleType | None = ...) -> xml.dom.minidom.Element: ...
```
Put the comment before the function it refers to

Reply:
Also, this could be typed precisely with a Protocol if you want.

Reply:
Thanks, I changed it use a Protocol.